bundle/edu.gemini.lchquery.servlet/src/main/java/edu/gemini/lchquery/servlet/LchQueryFunctor.java

```diff
@@ -1,549 +0,0 @@
-import edu.gemini.spModel.gemini.altair.InstAltair;
-import edu.gemini.spModel.gemini.obscomp.SPProgram;
-import edu.gemini.spModel.gemini.obscomp.SPSiteQuality;
-import edu.gemini.spModel.obs.ObsClassService;
-import edu.gemini.spModel.obs.ObservationStatus;
-import edu.gemini.spModel.obs.SPObservation;
-import edu.gemini.spModel.obs.SchedulingBlock;
-import edu.gemini.spModel.obs.context.ObsContext;
-import edu.gemini.spModel.obsclass.ObsClass;
-import edu.gemini.spModel.obscomp.SPInstObsComp;
-import edu.gemini.spModel.target.SPTarget;
-import edu.gemini.spModel.target.env.GuideGroup;
-import edu.gemini.spModel.target.env.GuideProbeTargets;
-import edu.gemini.spModel.target.env.TargetEnvironment;
-import edu.gemini.spModel.target.obsComp.TargetObsComp;
-import edu.gemini.spModel.too.Too;
-import edu.gemini.spModel.too.TooType;
-import edu.gemini.spModel.util.SPTreeUtil;
-import jsky.util.StringUtil;
-
-import java.io.Serializable;
-import java.rmi.RemoteException;
-import java.security.Principal;
-import java.text.SimpleDateFormat;
-import java.util.*;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
-/**
- * Implements the LCH query on the ODB
- */
-public class LchQueryFunctor extends DBAbstractQueryFunctor implements IDBParallelFunctor {
-    private static final Logger LOG = Logger.getLogger(LchQueryFunctor.class.getName());
-
-    private static final long MS_PER_SECOND = 1000;
-    private static final long MS_PER_MINUTE = MS_PER_SECOND * 60;
-    private static final long MS_PER_HOUR = MS_PER_MINUTE * 60;
-
-    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss z");
-
-    static {
-        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
-    }
-
-    // Type of query results: only programs, progs + observations or progs + obs + targets
-    enum QueryType {
-        PROGRAMS, OBSERVATIONS, TARGETS
-    }
-
-    private QueryResult _queryResult;
-    private QueryType _queryType;
-
-    private String _programSemester;
-    private String _programTitle;
-    private String _programReference;
-    private String _programActive;
-    private String _programCompleted;
-    private String _programNotifyPi;
-    private String _programRollover;
-
-    private String _observationTooStatus;
-    private String _observationName;
-    private String _observationStatus;
-    private String _observationInstrument;
-    private String _observationAo;
-    private String _observationClass;
-
-    public LchQueryFunctor(QueryType queryType, String programSemester, String programTitle, String programReference,
-                           String programActive, String programCompleted, String programNotifyPi, String programRollover,
-                           String observationTooStatus, String observationName, String observationStatus, String observationInstrument,
-                           String observationAo, String observationClass) {
```

Comment:
One of the main purposes of this PR was to get rid of horrible things like this...